flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/streaming/util/TestStreamEnvironment.java

```diff
@@ -92,10 +94,18 @@ public static void setAsContext(
                                 Duration.ofSeconds(2));
                     }
                     if (STATE_CHANGE_LOG_CONFIG.equalsIgnoreCase(STATE_CHANGE_LOG_CONFIG_ON)) {
-                        conf.set(CheckpointingOptions.ENABLE_STATE_CHANGE_LOG, true);
+                        if (isConfigurationSupportedByChangelog(miniCluster.getConfiguration())) {
+                            conf.set(CheckpointingOptions.ENABLE_STATE_CHANGE_LOG, true);
+                        }
                     } else if (STATE_CHANGE_LOG_CONFIG.equalsIgnoreCase(
                             STATE_CHANGE_LOG_CONFIG_RAND)) {
-                        randomize(conf, CheckpointingOptions.ENABLE_STATE_CHANGE_LOG, true, false);
+                        if (isConfigurationSupportedByChangelog(miniCluster.getConfiguration())) {
+                            randomize(
+                                    conf,
+                                    CheckpointingOptions.ENABLE_STATE_CHANGE_LOG,
+                                    true,
+                                    false);
+                        }
```

Comment:
This seems only to check `LOCAL_RECOVERY` which is configured in the mini-cluster level, since it checks against `minicluster.getConfiguration`. What if local recovery configured in other level, like locally?

Reply:
Good point, but local recovery currently can only be enabled in the cluster configuration.